### PR TITLE
Load weights on segmentation/train.py when using --resume and --test-only flags

### PR DESCRIPTION
**Pull request by datumbox**
A good way to verify the accuracy of a specific epoch checkpoint on the evaluation dataset is to execute:
```
python -m torch.distributed.launch --nproc_per_node=2 --use_env train.py --dataset coco --model model_name_here\ 
  --test-only --resume /path/to/checkpoint.pth
```

This is supported on both the classification and detection reference scripts but not on segmentation. More specifically on segmentation, the specified weights are not loaded and thus we report the accuracy of randomly initialized weights. This PR fixes this.

### CODE REVIEW

File: references/segmentation/train.py [hunk 1]
```diff
@@ -155,10 +150,16 @@ def main(args):
 
     if args.resume:
         checkpoint = torch.load(args.resume, map_location='cpu')
-        model_without_ddp.load_state_dict(checkpoint['model'])
-        optimizer.load_state_dict(checkpoint['optimizer'])
-        lr_scheduler.load_state_dict(checkpoint['lr_scheduler'])
-        args.start_epoch = checkpoint['epoch'] + 1
+        model_without_ddp.load_state_dict(checkpoint['model'], strict=not args.test_only)
```
Comment (datumbox) [inline references/segmentation/train.py:153]: `strict=False` required to avoid having to configure auxiliary classifiers

File: references/segmentation/train.py [hunk 2]
```diff
@@ -155,10 +150,16 @@ def main(args):
...
+        if not args.test_only:
+            optimizer.load_state_dict(checkpoint['optimizer'])
```
Comment (datumbox) [inline references/segmentation/train.py:155]: We avoid loading the weights of other objects if in test-only mode. Similar to the above, this is done to avoid having to handle auxiliary classifiers.

**Review by datumbox — COMMENTED** [review]
I left a few notes on the implementation:

**Review by fmassa — APPROVED** [review]
Looks great, thanks for improving this Vasilis!
